.github/workflows/build.yml

```diff
@@ -60,35 +128,140 @@ jobs:
 
           echo "result=$RESULT" >> $GITHUB_OUTPUT
 
-  build:
-    name: build (${{ matrix.config.os }})
+  build-type-determination:
     needs: run-determination
     if: needs.run-determination.outputs.result == 'true'
+    runs-on: ubuntu-latest
+    outputs:
+      is-release: ${{ steps.determination.outputs.is-release }}
+      is-nightly: ${{ steps.determination.outputs.is-nightly }}
+      channel-name: ${{ steps.determination.outputs.channel-name }}
```

Comment:
Would it make sense to pull up the `github.repository == 'arduino/arduino-ide'` check here into an `is-fork` (or similar) output?

Reply:
Sounds good. Resolved via https://github.com/arduino/arduino-ide/pull/2245/commits/d46869f0287381d2cb38c015737143c6f4c3ccc8 and improved via https://github.com/arduino/arduino-ide/pull/2245/commits/08cef1de2f305c5138149eef078811d087a4984a